### ISSUE TITLE
data collection issues

### ISSUE THREAD

**Issue by rax-rstark** [rax-rstark/playbooks#2]
Issue 1: Some devices do not have any information to report
Solution: Add check to only run collection & reporting tasks on hpe servers (some are Dell, which needs it's own report)

Issue 2: hostnames missing from C1, C2, C3 reports
Comments: 

- why gather facts if the fact is not reporting? 
- Maybe try running with hostname output registered to a variable and turn off fact gathering to save time?
- (some) missing hostnames do output tool versions

Issue 3: missing versions on hpe servers
comment: check manually, maybe out of date tooling in place? Different than Issue 1.

**Comment by rax-rstark** [comment]
re-run command while watching log file to spot issues:
```
root@696184-infra01:/opt/openstack-ansible/playbooks# ANSIBLE_STDOUT_CALLBACK=unixy ansible-playbook ~rack/playbooks/hpe-firmware-audit.yml
...

Get 696226-cinderSSD01 storage controller firmware version...
  696226-cinderSSD01 done | stdout: 6.60-0
6.60-0
6.60-0
Get 696226-cinderSSD01 iLO firmware version...
  696226-cinderSSD01 done | stdout: 2.61
Append 696226-cinderSSD01 details to local audit file...
  696226-cinderSSD01 -> localhost done | msg: line added```

The above output correlates to the below log file. We can see that the task pulled in more than we expected…

```
Hostname, HPE ROM Version, Controller Version, iLO Version
696185-infra02,v2.64 (10/17/2018) P89,6.60-0,2.61
696184-infra01,v2.64 (10/17/2018) P89,6.60-0,2.61
696187-infra04,v2.64 (10/17/2018) P89,6.60-0,2.61
696188-infra05,v2.64 (10/17/2018) P89,4.52-0,2.61
696186-infra03,v2.64 (10/17/2018) P89,6.60-0,2.61
696237-swiftstorage03,v2.64 (10/17/2018) P89,6.60-0,2.61
696239-swiftstorage05,v2.64 (10/17/2018) P89,6.60-0,2.61
696235-swiftstorage01,v2.76 (10/21/2019) P89,7.00-0,2.73
696238-swiftstorage04,v2.64 (10/17/2018) P89,6.60-0,2.61
696236-swiftstorage02,v2.64 (10/17/2018) P89,6.60-0,2.61
696232-cinderSAS03,P89 10/17/2018,6.60-0,2.61
696226-cinderSSD01,P89 10/17/2018,6.60-0
6.60-0
6.60-0,2.61
```

a manual check on the node shows the same:
```
root@696226-cinderSSD01:~# ssacli ctrl all show detail | awk  -F': ' '/Firmware/ { print $2 }'
6.60-0
6.60-0
6.60-0
```

Check the full output of the command shows we have 3 controllers in this one host. the line breaks in stdout were unaccounted for in output file.

**Comment by rax-rstark** [comment]
`ssacli ctrl all show detail | awk -F': ' '/Firmware/ { print $2 }' | tr '\n' ' '`

**Comment by rax-rstark** [comment]
Issue 2 & 3 resolved by #3

**Comment by rax-rstark** [comment]
Even though hostname issue was resolved, added #4 to make things run faster

**Comment by rax-rstark** [comment]
#6 speeds things up.
